apps/comps/rpc/router/user/link-wallet.ts

```diff
@@ -0,0 +1,73 @@
+import { ORPCError } from "@orpc/server";
+import { z } from "zod";
+
+import { verifyPrivyUserHasLinkedWallet } from "@recallnet/services/lib";
+
+import { base } from "@/rpc/context/base";
+import { authMiddleware } from "@/rpc/middleware/auth";
+import { serializeUser } from "@/rpc/router/utils/serialize-user";
+
+export const linkWallet = base
+  .use(authMiddleware)
+  .input(z.object({ walletAddress: z.string() }))
+  .handler(async ({ input, context, errors }) => {
+    try {
+      // Get Privy identity token from cookies
+      const idToken = context.cookies.get("privy-id-token")?.value;
```

Comment:
This is totally fine, but also could be nice to add the privy id token to the context at in the privy user middleware. Either way is fine, though.

Reply:
done. Wasn't totally sure if it made sense to make this function depend on both authMiddleware and privyUserMiddleware or pass the token through all the middleware layers so it was accessible just on the authMiddleware.  I wound up going with making linkWallet use both middlewares, lmk if you think it should work the other way